sky/data/sky_csync.py

```diff
@@ -0,0 +1,545 @@
+"""CSYNC module"""
+import functools
+import os
+import shlex
+import shutil
+import subprocess
+import sys
+import tempfile
+import time
+from typing import Any, List, Optional, Tuple
+
+import click
+import psutil
+
+from sky import exceptions
+from sky import sky_logging
+from sky.data import mounting_utils
+from sky.data.storage_utils import StorageMode
+from sky.skylet import constants
+from sky.utils import common_utils
+from sky.utils import db_utils
+
+logger = sky_logging.init_logger(__name__)
+
+_CSYNC_DB_PATH = os.path.join(constants.CSYNC_DIR, 'sky_csync.db')
+_CSYNC_READ_PATH = os.path.join(constants.CSYNC_DIR, 'read_{pid}')
+_CSYNC_WRITE_PATH = os.path.join(constants.CSYNC_DIR, 'write_{pid}')
+
+_BOOT_TIME = None
+_CURSOR = None
+_CONN = None
+_DB = None
+_MAX_SYNC_RETRIES = 10
+# Allowing minimal seconds of interval between SYNC calls.
+_MIN_SYNC_CALL_INTERVAL = 3
+_DEFAULT_S3_NUM_THREAD = 10
+CSYNC_DEFAULT_INTERVAL_SECONDS = 600
+
+
+def connect_db(func):
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        global _DB, _CURSOR, _CONN, _BOOT_TIME
+
+        def create_table(cursor, conn):
+            cursor.execute("""\
+                CREATE TABLE IF NOT EXISTS running_csync (
+                csync_pid INTEGER PRIMARY KEY,
+                sync_pid INTEGER DEFAULT -1,
+                storage_mount_pid INTEGER,
+                redirect_mount_pid INTEGER,
+                mountpoint_path TEXT,
+                boot_time FLOAT)""")
+
+            conn.commit()
+
+        if _DB is None:
+            db_path = os.path.expanduser(_CSYNC_DB_PATH)
+            os.makedirs(os.path.abspath(os.path.expanduser(
+                constants.CSYNC_DIR)),
+                        exist_ok=True)
+            _DB = db_utils.SQLiteConn(db_path, create_table)
+
+        _CURSOR = _DB.cursor
+        _CONN = _DB.conn
+        _BOOT_TIME = psutil.boot_time()
+
+        return func(*args, **kwargs)
+
+    return wrapper
+
+
+@connect_db
+def _add_running_csync(csync_pid: int, storage_mount_pid: int,
+                       redirect_mount_pid: int, mountpoint_path: str):
+    """Given the ids processes necessary to CSYNC, it creates a row with it"""
+    assert _CURSOR is not None
+    assert _CONN is not None
+    _CURSOR.execute(
+        'INSERT INTO running_csync '
+        '(csync_pid, storage_mount_pid, redirect_mount_pid, '
+        'mountpoint_path, boot_time) '
+        'VALUES (?, ?, ?, ?, ?)',
+        (csync_pid, storage_mount_pid, redirect_mount_pid, mountpoint_path,
+         _BOOT_TIME))
+    _CONN.commit()
+
+
+@connect_db
+def _get_all_running_csync_pid() -> List[Any]:
```

Comment:
```suggestion
def _get_all_running_csync_pid() -> List[int]:
```
is there any reason to use Any?